vulkan_generator/vulkan_parser/api/types.py

```diff
@@ -0,0 +1,638 @@
+from typing import Optional
+
+
+@dataclass
+class ExternalPlatform:
+    name: str
+    protect: str
+
+
+@dataclass
+class ExternalInclude:
+    header: str
+
+    # C directive to include the header, if given
+    directive: Optional[str]
+
+
+@dataclass
+class VulkanDefine:
+    # This is the key used by other Vulkan types to refer a define
+    key: str
+    name: str
+    value: str
+
+    # Is this define added after Vulkan 1.0 via extension or a later core version
+    extension: bool
+
+
+@dataclass
+class VulkanType:
+    """Base class for a Vulkan Type. All the other Types should inherit from this class"""
+    typename: str
+
+
+@dataclass
+class VulkanTypeAlias(VulkanType):
+    """Base class for a Vulkan Alias. All the other aliases should inherit from this class"""
+    aliased_type: VulkanType
+
+
+@dataclass
+class ExternalType(VulkanType):
+    """
+    The metadata defines a C type
+
+    This is actually not a Vulkan type but because its referenced from Vulkan types frequently
+    it's easier to pass it as a Vulkan type
+    """
+    # If any header is required to use this type
+    source_header: Optional[str]
+
+    # Is this type a standard C type
+    ctype: bool
+
+
+@dataclass
+class VulkanBaseType(VulkanType):
+    """Base class for a Vulkan basetype"""
+    # If there is no base type, then it is a forward declaration
+    basetype: Optional[str]
+
+
+@dataclass
+class VulkanHandle(VulkanType):
+    """The meta data defines a Vulkan Handle"""
+    # Melih TODO: Vulkan Handles have object type in the XML that might be required in the future
+    dispatchable: bool
+
+
+@dataclass
+class VulkanHandleAlias(VulkanTypeAlias):
+    """The meta data defines a Vulkan Handle alias"""
+
+
+@dataclass
+class VulkanEnumField:
+    """The meta data defines a value in an enum field"""
+    name: str
+
+    # Python3 int's are limitless
+    value: int
+
+    # Enum fields are sometimes represented hexadecimal especially in bitmasks.
+    # It can be pretty irregular in the spec. Therefore, we can explicitly
+    # state the representation when necessary.
+    representation: str
+
+    parent: VulkanEnum
+
+    # Print parent's typename only to prevent circular dependency
+    def to_str(self) -> str:
+        return f"""
+        {self.name}
+            Value: {self.value}
+            Representation: {self.representation}
+            Parent: {self.parent.typename}
+        """
+
+    def __repr__(self) -> str:
+        return self.to_str()
+
+    def __str__(self) -> str:
+        return self.to_str()
+
+
+@dataclass
+class VulkanEnumFieldAlias:
+    """The meta data defines a value in an alias to an enum field"""
+    name: str
+    aliased_field: VulkanEnumField
+
+    # Is this define added after Vulkan 1.0 via extension or a later core version
+    extension: bool
+
+    parent: VulkanEnum
+
+    # Print parent's typename only to prevent circular dependency
+    def to_str(self) -> str:
+        return f"""
```

Comment:
Anywhere you wrap an fstring across multiple lines like this, you should use dedent() to remove the whitespace added from "where the fstring is hosted in the code"